### PR TITLE
Update to Micrometer 1.5.1 LTS

### PR DESCRIPTION
**Pull request by anuraaga**
Uses the new distribution config methods that are compatible from 1.3 up to 1.5 and beyond.

Removes the fork of `ExecutorServiceMetrics`. I left out `isMonitoredExecutor` - it's not entirely clear whether we shouldn't add metrics again with our prefix even if a user already is monitoring the input executor. And anyways, double-wrapped callable in a very rare case doesn't seem worth using a library's internal API.

### CODE REVIEW

**Comment by trustin** [comment]
> I left out `isMonitoredExecutor` - it's not entirely clear whether we shouldn't add metrics again with our prefix even if a user already is monitoring the input executor.

Looks like it's removed, too?

File: core/src/main/java/com/linecorp/armeria/server/ServerConfig.java [hunk 1]
```diff
@@ -155,11 +155,9 @@
                                    gracefulShutdownQuietPeriod, "gracefulShutdownQuietPeriod");
 
         requireNonNull(blockingTaskExecutor, "blockingTaskExecutor");
-        if (!ExecutorServiceMetrics.isMonitoredExecutor(blockingTaskExecutor)) {
```
Comment (ikhoon) [inline core/src/main/java/com/linecorp/armeria/server/ServerConfig.java:158]: Could you revive this check only for `TimedScheduledExecutorService, TimedExecutorService and TimedExecutor`?
https://github.com/line/armeria/pull/2612#discussion_r395126261

Comment (anuraaga) [reply core/src/main/java/com/linecorp/armeria/server/ServerConfig.java:158]: Hmm? Any thoughts on what I wrote in the PR description?

Comment (ikhoon) [reply core/src/main/java/com/linecorp/armeria/server/ServerConfig.java:158]: Oops, I missed the PR description. 😅 That makes sense!

**Comment by minwoox** [comment]
https://ci.appveyor.com/project/line/armeria/builds/32677045/job/xlq6xh7f0rjycmm6#L2861
Oops, it seems like we should upgrade the version of Spring too.

File: spring/boot1-autoconfigure/build.gradle [hunk 2]
```diff
@@ -1,19 +1,24 @@
...
     compileOnly project(':thrift')
     implementation project(':logback')
 
-    [ 'micrometer-core', 'micrometer-spring-legacy' ].each {
-        implementation("io.micrometer:$it") {
-            version {
-                // Will fail the build if the override doesn't work
-                strictly MICROMETER_VERSION
-            }
+    implementation("io.micrometer:micrometer-spring-legacy") {
```
Comment (anuraaga) [inline spring/boot1-autoconfigure/build.gradle:9]: @shakuzen Any thoughts on mixing LTS versions like this?

Comment (shakuzen) [reply spring/boot1-autoconfigure/build.gradle:9]: It's probably better to keep the version consistent within this sub-project. I would expect a user still using `micrometer-spring-legacy` to stay on Micrometer 1.3.x. Of course the best suggestion is that they upgrade to Spring Boot 2 to be on the latest supported versions, though.

Comment (anuraaga) [reply spring/boot1-autoconfigure/build.gradle:9]: Hmm - in that case can we backport the `ExecutorServiceMetrics` change to 1.3.x? Right now we have the conundrum of having it only in 1.5.x, but only a 1.3.x of spring-legacy.

**Comment by anuraaga** [comment]
@shakuzen Thanks for backporting so quickly!

This PR should be good to go now

**Comment by trustin** [comment]
Thanks, @anuraaga, @shakuzen et al.

**Pull request merged**